File: Data Structures & Algorithms/maximum-subarray/submission-4.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_global = -float('inf')
        change = 0
        for n in nums:
            change = max(n, change + n)
            max_global = max(max_global, change)
        
        return max_global
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const set = new Set(nums);
        let maxLen = 0;
        set.forEach(s => {
            let n = s;
            if (!set.has(n - 1)) {
                // n - 1 is not in the set, then s is the starting point
                let currLen = 1;
                while (set.has(n + 1)) {
                    currLen ++;
                    n ++;
                }
                maxLen = Math.max(maxLen, currLen)
            }
        })

        return maxLen
    }

}
